extensions/amp-viewer-integration/0.1/messaging.js

```diff
@@ -19,9 +19,29 @@ import {listen} from '../../../src/event-helper';
 import {dev} from '../../../src/log';
 
 const TAG = 'amp-viewer-messaging';
-const sentinel_ = '__AMPHTML__';
-const requestSentinel_ = sentinel_ + 'REQUEST';
-const responseSentinel_ = sentinel_ + 'RESPONSE';
+const APP = '__AMPHTML__';
+
+/**
+ * @enum {string}
+ * @private
+ */
+const MessageType_ = {
+  REQUEST: 'q',
+  RESPONSE: 's',
+};
+
+/**
+ * @typedef {{
+ *   app: !string,
+ *   type: !string,
+ *   requestid: !number,
+ *   name: !string,
+ *   data: *,
+ *   rsvp: ?boolean,
+ *   error: *
```

Comment:
Let's spell out what we expect here explicitly. Is this a string for now? If so, let's do `error: (string|undefined)`.